### ISSUE TITLE
be more careful about leaking Activity references

### ISSUE THREAD

**Issue by durka** [durka/HallMonitor#20]
We might be fine... but some classes such as Functions and CameraHelper do have references to the DefaultActivity. These may or may not need to be refactored to avoid leaks. Source: idle conversations on IRC and http://www.curious-creature.org/2008/12/18/avoid-memory-leaks-on-android/

**Comment by Wallace0** [comment]
Yep, well I think we need a more general bug fix and tidy up anyway - to my mind I think we are about there in terms of basic capabilities for a v1 release - definitely if we also include the 2 phased control thing as well.

**Comment by manusfreedom** [comment]
@durka Normally it's better since 2.0.0
